course4/homework1/vector2d.h

```diff
@@ -1,11 +1,41 @@
 #pragma once
 
+#include <cmath>
 #include <ostream>
 
 struct Vector2d
 {
 public:
-    Vector2d(int x, int y) : mX(x), mY(y) {}
+    // named ctors
+    static Vector2d fromCarthesian(const double x, const double y) {
+        return { x, y };
+    }
+    static Vector2d fromPolar(const double rho, const double phi) {
+        return { rho * cos(phi), rho * sin(phi) };
+    }
+    // ctor
+    Vector2d(double x, double y) : mX(x), mY(y) {}
+    // setters
+    void X(const double x) {
+        mX = x; 
+    }
+    void Y(const double y) {
+        mY = y;
+    }
+    // getters 
+    double X() const {
+        return mX;
+    }
+    double Y() const {
+        return mY;
+    }
+    double Rho() const {
+        return sqrt(mX*mX + mY*mY);
```

Comment:
Use ```std::sqrt``` ans ```std::atan2```. The old, non-qualified, c-style functions still happen to work, but this may change in the future.